src/metabase/driver/h2.clj

```diff
@@ -86,70 +90,45 @@
     ;; For :native queries check to make sure the DB in question has a (non-default) NAME property specified in the
     ;; connection string. We don't allow SQL execution on H2 databases for the default admin account for security
     ;; reasons
-    (when (= (keyword query-type) :native)
-      (let [{:keys [details]} (qp.store/database)
-            user              (db-details->user details)]
-        (when (or (str/blank? user)
-                  (= user "sa"))        ; "sa" is the default USER
-          (throw
-           (ex-info (tru "Running SQL queries against H2 databases using the default (admin) database user is forbidden.")
-             {:type qp.error-type/db})))))))
-
-(defn- make-h2-parser [h2-db-id]
+           (when (= (keyword query-type) :native)
+             (let [{:keys [details]} (qp.store/database)
+                   user              (db-details->user details)]
+               (when (or (str/blank? user)
+                         (= user "sa"))        ; "sa" is the default USER
+                 (throw
+                  (ex-info (tru "Running SQL queries against H2 databases using the default (admin) database user is forbidden.")
+                           {:type qp.error-type/db})))))))
+
+(defn- get-field
+  "Returns value of private field"
+  [obj field]
+  (.get (doto (.getDeclaredField (class obj) field)
+          (.setAccessible true))
+        obj))
```

Comment:
I know you didn't write this stuff but it makes me pretty itchy to see code circumventing encapsulation thru reflection hacks that don't have so much as a single comment explaining why they're doing it

It might be worth writing a little blurb someone around here just to explain what's going on so people doing need to scan like 60 lines of code to figure out what the point of all this is